src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,150 @@
+                }
+            }
+
+        } else if (fullCommand.contains("mark")) {
+            if (fullCommand.substring(4).isBlank()) {
+                throw new DukeException("More information is required to mark a task");
+            } else {
+                String[] unparsedTaskIndex = fullCommand.split(" ");
+                int taskIndex = Integer.parseInt(unparsedTaskIndex[1]) - 1;
+                if (taskIndex >= (taskList.size()) || taskIndex < 0) {
+                    throw new DukeException("There are only " + taskList.size() + " tasks");
+                } else {
+                    return new MarkCommand(taskIndex);
+                }
+            }
+
+        } else if (fullCommand.contains("delete")) {
+            if (fullCommand.substring(6).isBlank()) {
+                throw new DukeException("More information is required to delete a task");
+            } else {
+                String[] unparsedTaskIndex = fullCommand.split(" ");
+                int taskIndex = Integer.parseInt(unparsedTaskIndex[1]) - 1;
+                if (taskIndex >= (taskList.size()) || taskIndex < 0) {
+                    throw new DukeException("There are only " + taskList.size() + " tasks");
+                } else {
+                    return new DeleteCommand(taskIndex);
+                }
+            }
+
+        } else if (fullCommand.contains("todo")) {
+            String taskInfo = fullCommand.substring(4);
+            if (taskInfo.isBlank()) {
+                throw new DukeException("The description of a todo cannot be empty");
+            } else {
+                String[] splitCommands = fullCommand.split(" ");
+                Task task = new ToDo(splitCommands[1]);
+                return new AddCommand(task);
+            }
+
+        } else if (fullCommand.contains("deadline")) {
+            String taskInfo = fullCommand.substring(8);
+            if (taskInfo.isBlank() || !fullCommand.contains("/by")
+                    || !fullCommand.contains("deadline ")
+                    || fullCommand.substring(fullCommand.indexOf("/by") + 3).isBlank()) {
+                throw new DukeException("The description of deadline needs more information");
+            } else {
+                String[] splitCommands = fullCommand.split(" /by ");
+                String[] taskName = splitCommands[0].split(" ");
+                Task task = new Deadline(taskName[1], Parser.parseDateTime(splitCommands[1]));
+                return new AddCommand(task);
+            }
+
+        } else if (fullCommand.contains("event")) {
+            String taskInfo = fullCommand.substring(5);
+            if (taskInfo.isBlank() || !fullCommand.contains("/from")
+                    || !fullCommand.contains("/to") || !fullCommand.contains("event ")
+                    || fullCommand.substring(fullCommand.indexOf("/from") + 5).isBlank()
+                    || fullCommand.substring(fullCommand.indexOf("/to") + 3).isBlank()) {
```

Comment:
could there be a way to shorten this expression ? maybe you can attempt to try creating the event and catching any exceptions thrown?